decam_process.py

```diff
@@ -15,324 +15,378 @@
     '''
+
+    def flatBiasIngest(repo, calibrepo, calibdatafiles):
+        '''
+        Ingest DECam flats and biases
+    
+        Parameters
+        ----------
+        repo: `str`
+            The output repository location on disk where ingested raw images live.
+        calibrepo: `str`
+            The output repository location on disk where ingested calibration images live.
+        calibdatafiles: `list`
+            A list of the filenames of each flat and bias image file.
+    
+        '''
+        print('Ingesting flats and biases...')
+        args = [repo, '--calib', calibrepo, '--mode', 'link', '--validity', '999']
+        args.extend(calibdatafiles)
+        argumentParser = IngestCalibsArgumentParser(name='ingestCalibs')
+        config = IngestCalibsConfig()
+        config.parse.retarget(ingestCalibs.DecamCalibsParseTask)
+        ingestTask = IngestCalibsTask(config=config, name='ingestCalibs')
+        parsedCmd = argumentParser.parse_args(config=config, args=args)
+        try:
+            ingestTask.run(parsedCmd)
+        except sqlite3.IntegrityError as detail:
+            print('sqlite3.IntegrityError: ', detail)
+            print('(sqlite3 doesn\'t think all the calibration files are unique)')
+        else:
+            print('Success!')
+            print('Calibrations corresponding to {0} are now ingested in {1}'.format(repo, calibrepo))
+        return
+
+    def defectIngest(repo, calibrepo, defectfiles):
+        '''
+        Ingest DECam defect images
+    
+        Parameters
+        ----------
+        repo: `str`
+            The output repository location on disk where ingested raw images live.
+        calibrepo: `str`
+            The output repository location on disk where ingested calibration images live.
+        defectfiles: `list`
+            A list of the filenames of each defect image file.
+            The first element in this list must be the name of a .tar.gz file
+            which contains all the compressed defect images.
+
+        This function assumes very particular things about defect ingestion:
+        - They must live in a .tar.gz file in the same location on disk as the other calibs
+        - They will be ingested using ingestCalibs.py run from the calibrepo directory
+        - They will be manually uncompressed and saved in calibrepo/defects/<tarballname>/.
+        - They will be added to the calib registry, but not linked like the flats and biases
+        '''
+        os.chdir(calibrepo)
+        try:
+            os.mkdir('defects')
+        except OSError:
+            # defects directory already exists
+            print('Defects were previously ingested, skipping...')
+        else:
+            print('Ingesting defects...')
+            defectargs = ['../../' + repo, '--calib', '.', '--calibType', 'defect',
+                          '--mode', 'skip', '--validity', '999']
+            defecttarball = defectfiles[0] + '.tar.gz'
```

Comment:
Should these names be camel-case or underscore-separated? PEP8 is kind of vague about local variables.